### PR TITLE
Add more checks to scheduleWarmUpFrame

### PR DESCRIPTION
**Pull request by xster**
Prevents crash from calling runApp twice. 

I'm not sure how to test it since AutomatedTestWidgetsFlutterBinding overrides it.

### CODE REVIEW

File: packages/flutter/lib/src/scheduler/binding.dart [hunk 1]
```diff
@@ -669,9 +669,14 @@ abstract class SchedulerBinding extends BindingBase with ServicesBinding {
   /// If a frame has already been scheduled with [scheduleFrame] or
   /// [scheduleForcedFrame], this call may delay that frame.
   ///
+  /// If any scheduled frame has already begun or if another
+  /// [scheduleWarmUpFrame] was already called, this call will be ignored.
+  ///
   /// Prefer [scheduleFrame] to update the display in normal operation.
   void scheduleWarmUpFrame() {
-    assert(!_warmUpFrame);
+    if (_warmUpFrame && !(schedulerPhase == SchedulerPhase.idle))
```
Comment (Hixie) [inline packages/flutter/lib/src/scheduler/binding.dart]: prefer `!=` over `!( == )`

Comment (Hixie) [reply packages/flutter/lib/src/scheduler/binding.dart]: also i'm pretty sure you want `||` not `&&` :-)

**Comment by Hixie** [comment]
you could test this by not using the widgets test framework, but instead creating your own TestSchedulerBinding (that mixes in SchedulerBinding) and driving that.

File: packages/flutter/lib/src/scheduler/binding.dart (hunk 1 above)
Comment (xster) [reply packages/flutter/lib/src/scheduler/binding.dart]: T.T

**Comment by xster** [comment]
Added a test

File: packages/flutter/test/scheduler/scheduler_test.dart [hunk 2]
```diff
@@ -84,4 +89,25 @@ void main() {
...
+          // Don't actually run the tasks, just record that it was scheduled.
+          timerQueueTasks.add(f);
+          return null;
+        },
+      ),
+    );
+
+    // A single call to scheduleWarmUpFrame queues up 2 Timer tasks.
+    expect(timerQueueTasks.length, 2);
+  });
```
Comment (Hixie) [inline packages/flutter/test/scheduler/scheduler_test.dart]: nice!

**Comment by Hixie** [comment]
LGTM

**Pull request merged**